src/NHibernate/Cache/StandardQueryCache.cs

```diff
@@ -335,9 +371,14 @@ public void Destroy()
 			ICacheAssembler[] returnTypes,
 			ISessionImplementor session,
 			IList result,
-			long ts)
+			long ts, string[] aliases)
 		{
-			var cacheable = new List<object>(result.Count + 1) { ts };
+			var cacheable =
+				new List<object>(result.Count + 1)
+				{
+					aliases == null ? ts : new object[] { ts, aliases }
```

Comment:
I have still to add tests for the JsonSerializer, for checking if a `string[]` will be supported or will have to turned into an `object[]`.